pkg/record/record.go

```diff
@@ -1,72 +1,136 @@
+	CallNumbers            []string          `json:"call_numbers,omitempty"`
+	Citation               string            `json:"citation,omitempty"`
+	ContentType            []string          `json:"content_type,omitempty"`
+	Contents               []string          `json:"contents,omitempty"`
+	Contributors           []*Contributor    `json:"contributors,omitempty"`
+	Dates                  []*Date           `json:"dates,omitempty"`
+	Edition                string            `json:"edition,omitempty"`
+	FileFormats            []string          `json:"file_formats,omitempty"`
+	Format                 string            `json:"format,omitempty"`
+	FundingInformation     []*Funding        `json:"funding_information,omitempty"`
+	Holdings               []*Holding        `json:"holdings,omitempty"`
+	Identifiers            []*Identifier     `json:"identifiers,omitempty"`
+	Languages              []string          `json:"languages,omitempty"`
+	Links                  []Link            `json:"links,omitempty"`
+	LiteraryForm           string            `json:"literary_form,omitempty"`
+	Locations              []*Location       `json:"locations,omitempty"`
+	Notes                  []*Note           `json:"notes,omitempty"`
+	Numbering              string            `json:"numbering,omitempty"`
+	PhysicalDescription    string            `json:"physical_description,omitempty"`
+	PublicationFrequency   []string          `json:"publication_frequency,omitempty"`
+	PublicationInformation []string          `json:"publication_information,omitempty"`
+	RelatedItems           []*RelatedItem    `json:"related_items,omitempty"`
+	Rights                 []*Right          `json:"rights,omitempty"`
+	Source                 string            `json:"source"`
+	SourceLink             string            `json:"source_link"`
+	Subjects               []*Subject        `json:"subjects,omitempty"`
+	Summary                []string          `json:"summary,omitempty"`
+	TimdexRecordId         string            `json:"timdex_record_id"`
+	Title                  string            `json:"title"`
+}
+
+// AlternateTitle object
+type AlternateTitle struct {
+	Kind  string `json:"kind,omitempty"`
 	Value string `json:"value"`
 }
 
-// RelatedItem is a port of a Record
-type RelatedItem struct {
-	Kind  string   `json:"kind"`
-	Value []string `json:"value"`
+// Contributor object
+type Contributor struct {
+	Affiliation   string `json:"affiliation,omitempty"`
+	Kind          string `json:"kind,omitempty"`
+	Identifier    string `json:"identifier,omitempty"`
+	MitAffiliated bool   `json:"mit_affiliated,omitempty"`
+	Value         string `json:"value"`
 }
 
-// Link is a port of a Record
-type Link struct {
-	Kind         string `json:"kind,omitempty"`
-	Text         string `json:"text,omitempty"`
-	URL          string `json:"url"`
-	Restrictions string `json:"restrictions,omitempty"`
+// Date object
+type Date struct {
```

Comment:
Yeah I think the API would have to handle querying both. Regardless, I'm not sure there's a better way to index the data since we know we have both single dates and date ranges.